### PR TITLE
kde4-*.eclass: Migrate to L10N flags

### PR DESCRIPTION
**Pull request by mgorny**

### CODE REVIEW

**Comment by mgorny** [comment]
@gentoo/kde, @ulm, it is ready!

**Comment by mgorny** [comment]
(or not yet)

**Comment by kensington** [comment]
@a17r

**Comment by a17r** [comment]
This is missing one crucial part, the actual migration of ``kde-apps/kde4-l10n`` to L10N, since ``kde-apps/kde-l10n`` is inheriting kde5.eclass, which is btw also using LINGUAS right now. Further, there are ``kde-apps/kdepim-l10n`` and ``kde-apps/ktp-l10n`` that should get the same treatment.

That's why I'd like to get https://github.com/gentoo/gentoo/pull/1749 merged first, to minimise the amount of ebuilds to work on.

File: eclass/kde4-functions.eclass [hunk 1]
```diff
@@ -376,7 +387,7 @@ _enable_selected_linguas_dir() {
 	done
 
 	for lingua in ${KDE_LINGUAS}; do
-		if use linguas_${lingua} ; then
+		if use "linguas_$(kde4_lingua_to_l10n ${lingua})" ; then
```
Comment (a17r) [inline eclass/kde4-functions.eclass]: l10n?

File: eclass/kde4-functions.eclass [hunk 2]
```diff
@@ -213,7 +224,7 @@ enable_selected_doc_linguas() {
 			# Add requested translations
 			local lingua
 			for lingua in en ${KDE_LINGUAS}; do
-				if [[ ${lingua} = en ]] || use linguas_${lingua}; then
+				if [[ ${lingua} = en ]] || use "l10n_$(kde4_lingua_to_l10n "${lingua}")"; then
```
Comment (a17r) [inline eclass/kde4-functions.eclass]: two quotes too many?

File: eclass/kde4-functions.eclass (hunk 1 above)
Comment (mgorny) [inline eclass/kde4-functions.eclass]: Good catch, thanks.

File: eclass/kde4-functions.eclass (hunk 2 above)
Comment (mgorny) [inline eclass/kde4-functions.eclass]: Nah, it guarantees random stuffs won't case this to go multi-arg ;-P.

Comment (a17r) [inline eclass/kde4-functions.eclass]: ah, k

**Comment by mgorny** [comment]
Thanks for the tips. I didn't have time to figure out which ebuilds do L10N on their own. I'll rebase this on your PR, and fix the remaining ebuilds you mentioned. Hopefully that'd be good enough to proceed.

Lemme know if you need someone to merge that other PR for you.

File: kde-apps/kde-l10n/kde-l10n-15.12.3-r2.ebuild [hunk 3]
```diff
@@ -34,22 +34,20 @@ MY_LANGS="ar bg bs ca ca@valencia cs da de el en_GB eo es et eu fa fi fr ga gl
...
 pkg_setup() {
 	if [[ -z ${A} ]]; then
 		elog
-		elog "You either have the LINGUAS variable unset, or it only"
-		elog "contains languages not supported by ${P}."
-		elog "You won't have any additional language support."
+		elog "None of the requested L10N are supported by ${P}."
 		elog
 		elog "${P} supports these language codes:"
 		elog "${MY_LANGS}"
```
Comment (a17r) [inline kde-apps/kde-l10n/kde-l10n-15.12.3-r2.ebuild]: this output is going to be inconsistent with the values users have to set in L10N

File: kde-apps/kde4-l10n/kde4-l10n-16.04.2.ebuild [hunk 4]
```diff
@@ -117,7 +115,7 @@ EOF
 		while read path; do
 			if use test ; then	# build a report w/ LINGUAS="*" to submit @upstream
```
Comment (a17r) [inline kde-apps/kde4-l10n/kde4-l10n-16.04.2.ebuild]: let's fix comments as well ;)

File: kde-apps/kde4-l10n/kde4-l10n-16.04.2.ebuild [hunk 5]
```diff
@@ -139,7 +137,7 @@ EOF
...
```
Comment (a17r) [inline kde-apps/kde4-l10n/kde4-l10n-16.04.2.ebuild]: same as above

**Comment by a17r** [comment]
Did you check kde5.eclass as well? It uses LINGUAS for po file and handbook trimming.

**Comment by a17r** [comment]
I've built all the 16.04.2 packages successfully, also checked with a ca-valencia and special case sr, seems to work fine for me

+1

File: kde-apps/kde-l10n/kde-l10n-15.12.3-r2.ebuild (hunk 3 above)
Comment (mgorny) [inline kde-apps/kde-l10n/kde-l10n-15.12.3-r2.ebuild]: Oh, why are you so through! I hoped you wouldn't notice and someone else would have to fix it later ;-P.

**Comment by mgorny** [comment]
Comments and some other nonsense updated.

What's the deal with KDE5 anyway? I don't see any kde5 ebuild setting L10N. If they're in for implicit LINGUAS use, then it should stay with LINGUAS.

**Comment by a17r** [comment]
Yes it's good, wasn't sure at first if that needed work as well.

File: kde-apps/kde-l10n/kde-l10n-15.12.3-r2.ebuild [hunk 6]
```diff
@@ -100,7 +98,7 @@ EOF
 	if [[ -d ${KHC}/po ]] ; then
 		pushd ${KHC}/po > /dev/null || die
 		for lang in *; do
-			if [[ -d ${lang} ]] && ! has ${lang} ${LINGUAS} ; then
+			if [[ -d ${lang} ]] && ! has ${lang} ${L10N} ; then
```
Comment (a17r) [inline kde-apps/kde-l10n/kde-l10n-15.12.3-r2.ebuild]: this is looping over linguas directories, so second occurence of ${lang} will need the substitution here as well

**Comment by a17r** [comment]
thanks for working on the ebuilds in detail! ;)

**Comment by johu** [comment]
👍 lgtm

**Pull request closed without merging**